src/test/url-loader/package-url-resolver_test.ts

```diff
@@ -152,6 +152,27 @@ suite('PackageUrlResolver', function() {
       assert.equal(
           resolver.resolve(fileRelativeUrl`%><><%=`, packageRoot), undefined);
     });
+
+    test('resolves a URL with no pathname', () => {
+      const foo = rootedFileUrl`1/2/foo.html?baz#bat`;
+      const bar = rootedFileUrl`1/2/bar.html`;
+      assert.equal(
+          resolver.resolve(fileRelativeUrl``, foo),
+          rootedFileUrl`1/2/foo.html?baz`);
```

Comment:
why is the fragment missing here?

Reply:
Fragments can never be implicit; there's no URL form you can construct relative to a URL with a fragment that will preserve the fragment.  So it is clearest to test that assumption here with an empty relative URL.